common/src/java/org/apache/hadoop/hive/common/type/Date.java

```diff
@@ -84,7 +84,7 @@ public class Date implements Comparable<Date> {
   private static final LocalDate EPOCH = LocalDate.of(1970, 1, 1);
 
   private static final DateTimeFormatter PARSE_FORMATTER =
-      new DateTimeFormatterBuilder().appendValue(YEAR, 1, 10, SignStyle.NORMAL).appendLiteral('-')
+      new DateTimeFormatterBuilder().appendValue(YEAR, 1, 4, SignStyle.NORMAL).appendLiteral('-')
```

Comment:
Years must be not 4 characters long, since date 1-1-1, 45-10-20, 457-05-16, 896-08-01 are valid dates, and as we allow the month and days to be one or two characters long it would be strange to say, years must be at least 4 character long, but it can be 10 character long. Most of the programming languages support dates up to years 9999 or even less, so we are good with that limit.